File: dp-core/src/main/java/com/dataplatform/system/entity/LoginLog.java
package com.dataplatform.system.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 登录日志实体
 */
@Data
public class LoginLog {
    private Long id;
    private String username;
    private String ipAddress;
    private String userAgent;
    private String browser;
    private String os;
    /** success / failure */
    private String status;
    private String message;
    private LocalDateTime loginTime;
}

File: dp-ui/src/api/system/loginLog.ts
import request from '../request'

export interface LoginLog {
  id: number
  username: string
  ipAddress: string
  userAgent: string
  browser: string
  os: string
  status: string
  message: string
  loginTime: string
}

/**
 * 分页查询登录日志
 */
export const getLoginLogList = (params: {
  page?: number
  pageSize?: number
  username?: string
  status?: string
  ipAddress?: string
}) => {
  return request.get('/login-log/list', { params })
}

/**
 * 清理历史登录日志
 */
export const cleanLoginLog = (days: number) => {
  return request.delete('/login-log/clean', { data: { days } })
}
